libqtile/group.py

```diff
@@ -30,12 +30,90 @@
 
 import xcffib
 import xcffib.xproto
+from xcffib.xproto import StackMode
 
+from enum import Flag
 from libqtile import hook, utils, window
 from libqtile.command_object import CommandObject
 from libqtile.log_utils import logger
 
 
+class _History:
+    """ Helper class to keep track of a history of objects. E.g. focused windows,
+    z-layers etc.
+    Elements can be added, removed, pushed to the top and the bottom. Lists
+    representing the current overall state can be requested.
+    For windows, floating, normal and maximized can be distinguished.
+    """
+
+    def __init__(self):
+        self.clients = list()
+
+    #     def insert(self, index, client):
+    #         index = max(len(self.clients), min(0, index))
+    #         if client in self.clients:
+    #             already_at = self.clients.index(client)
+    #             self.clients.remove(client)
+    #             if already_at < index:
+    #                 index = index - 1
+    #         self.clients.insert(index, client)
+    #         return index
```

Comment:
I guess we can drop this?

Reply:
Don't forget this is WIP. :) I hope that the entire commented method can be removed, but for now I just wanted to keep it around.